File: packages/deterministic-mapping/src/mappings/diaper.ts
import { DIMENSIONS, DOMAINS, INTENTION_TYPES, CanonicalFact, MappingResult } from '../types';

export function mapDiaperFacts(facts: CanonicalFact[]): MappingResult | null {
  const diaperFact = facts.find((f) => f.dimension === DIMENSIONS.DIAPER_CHANGE_TYPE);
  if (!diaperFact) return null;

  return {
    domain: DOMAINS.DIAPER,
    type: INTENTION_TYPES.DIAPER_CHANGE,
    attributes: { changeType: diaperFact.value },
    confidence: diaperFact.confidence,
  };
}

File: packages/mcp-intent-gateway/src/mcp_intent_gateway/mapping/diaper.py
"""Diaper mapping function."""

from typing import Any

from ..domain.constants import Dimension, Domain, IntentionType
from ..models.canonical_fact import CanonicalFact


def map_diaper_facts(facts: list[CanonicalFact]) -> dict[str, Any] | None:
    """
    Map diaper-related facts to a MappingResult.

    Args:
        facts: List of canonical facts to process

    Returns:
        Mapping result dict or None if no diaper facts found
    """
    diaper_fact = next(
        (f for f in facts if f.dimension == Dimension.DIAPER_CHANGE_TYPE),
        None,
    )

    if not diaper_fact:
        return None

    return {
        "domain": Domain.DIAPER,
        "type": IntentionType.DIAPER_CHANGE,
        "attributes": {"changeType": diaper_fact.value},
        "confidence": diaper_fact.confidence,
    }
